Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -681,57 +672,90 @@ void ABenchmarkGymGameModeBase::OnRepTotalNPCs()
 	OnTotalNPCsUpdated(TotalNPCs);
 }
 
-int32 ABenchmarkGymGameModeBase::GetActorClassCount(TSubclassOf<AActor> ActorClass) const
+void ABenchmarkGymGameModeBase::OnActorCountReportId()
 {
-	const UCounterComponent* CounterComponent = Cast<UCounterComponent>(GameState->GetComponentByClass(UCounterComponent::StaticClass()));
-	check(CounterComponent != nullptr)
-
-	return CounterComponent->GetActorClassCount(ActorClass);
+	UpdateAndReportActorCounts();
 }
 
-void ABenchmarkGymGameModeBase::SetLifetime(int32 Lifetime)
+void ABenchmarkGymGameModeBase::UpdateAndReportActorCounts()
 {
-	if (TestLifetimeTimer.SetTimer(Lifetime))
+	const UWorld* World = GetWorld();
+	bool bSpatialEnabled = USpatialStatics::IsSpatialNetworkingEnabled();
+	const FString WorkerID = bSpatialEnabled ? GetGameInstance()->GetSpatialWorkerId() : TEXT("Worker1");
+	if (WorkerID.IsEmpty())
 	{
-		TestLifetimeTimer.SetLock(true);
-		UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Setting NFR test lifetime %d"), Lifetime);
+		return;
```

Comment:
This seems like a fatal also

Reply:
Only reason this was here was that I found the GetGameInstance()->GetSpatialWorkerId() was not set early in the deployment.

Reply:
Is it worth logging just in case?

Reply:
Sure.